From e4cea21b5f5da00cbd4b589ab5f849404ce4987a Mon Sep 17 00:00:00 2001
From: Konstantin Khlebnikov <khlebnikov@openvz.org>
Date: Fri, 27 Mar 2009 15:03:57 +0300
Subject: [PATCH 78/82] VE: fix idle time accounting

Make both account ways simmetic: idle time accounted as idle or iowait,
depending on number tasks in iowait state.

http://bugzilla.openvz.org/show_bug.cgi?id=1217
(#114633)

Signed-off-by: Konstantin Khlebnikov <khlebnikov@openvz.org>
Signed-off-by: Pavel Emelyanov <xemul@openvz.org>
---
 kernel/sched.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/kernel/sched.c b/kernel/sched.c
index 2a86a05..22942a3 100644
--- a/kernel/sched.c
+++ b/kernel/sched.c
@@ -1026,7 +1026,7 @@ cycles_t ve_sched_get_idle_time(struct ve_struct *ve, int cpu)
 		v = read_seqcount_begin(&ve_stat->stat_lock);
 		ret = ve_stat->idle_time;
 		strt = ve_stat->strt_idle_time;
-		if (strt && nr_uninterruptible_ve(ve) == 0) {
+		if (strt && nr_iowait_ve(ve) == 0) {
 			cycles = get_cycles();
 			if (cycles_after(cycles, strt))
 				ret += cycles - strt;
-- 
1.6.2.2

